### PR TITLE
Draft1 review revision

### PR DESCRIPTION
**Pull request by sbarnum**

### CODE REVIEW

File: examples/message.json [hunk 1]
```diff
@@ -109,10 +134,10 @@
       "propertyBundle": [
         {
           "@type": "Message",
+          "application": "application2",
```
Comment (casework) [inline examples/message.json]: Should message traces be agnostic to which application they came from in exchange for having relationships instead?

File: examples/message.json [hunk 2]
```diff
@@ -109,10 +134,10 @@
...
           "body": "I said some things in a tweet! @sarahsmithtweeter #hashtag",
-          "sender": "account3",
-          "participant": [
-            "account3",
+          "from": "account3",
+          "to": [
```
Comment (casework) [inline examples/message.json]: We should get a final consensus from the group whether to change these to "to" and "from".

File: examples/forensic_lifecycle.json [hunk 3]
```diff
@@ -63,50 +63,24 @@
...
+        "annotation6", "forensic_action6", "provenance_record1", "provenance_record2", "provenance_record3",
+        "provenance_record4", "provenance_record5", "provenance_record6", "provenance_record7", "provenance_record8",
+        "provenance_record9", "provenance_record10", "provenance_record11", "provenance_record12", "provenance_record13",
+        "provenance_record14", "provenance_record15", "provenance_record16", "provenance_record17", "provenance_record18",
+        "provenance_record19", "provenance_record20", "provenance_record21", "device1", "sd_card1", "android_image",
+        "sd_card1_image", "chat_messages_report", "plaso_storage_file", "os1", "attachment_file", "message_database",
+        "thumbnail_database", "image_partition", "message_action1", "thread1", "message1", "location1", "account1",
+        "account2", "decoded_blob", "decrypted_blob", "sqlite_blob", "tool1", "tool2", "tool3", "fedex_dropoff_location1",
+        "forensic_lab1", "forensic_lab_computer1", "config_file", "log_file", "windows_registries_report", "parser1",
+        "attachment_file"
```
Comment (casework) [inline examples/forensic_lifecycle.json]: Don't we loose the order the forensic actions were performed if we do this?
Also, wouldn't this object become immensely large if we do this? I though each forensic action holds the provenance records which in turn holds the objects that were created.

File: examples/forensic_lifecycle.json [hunk 4]
```diff
@@ -116,15 +90,20 @@
...
-      "result": [
-        "provenance_record1"
+      "propertyBundle": [
+        {
+          "@type": "ActionRefs",
+          "performer": "investigator1",
+          "result": [
+            "provenance_record1"
+          ]
+        }
```
Comment (casework) [inline examples/forensic_lifecycle.json]: See other comments about ActionRefs.

File: examples/forensic_lifecycle.json [hunk 5]
```diff
@@ -116,15 +90,20 @@
...
       ]
     },
     {
       "@id": "lifecycle_phase1",
       "@type": "Relationship",
-      "source": "forensic_action1",
+      "object": "forensic_action1",
```
Comment (casework) [inline examples/forensic_lifecycle.json]: I thought we had "source" and "target" for Relationship objects...

File: examples/forensic_lifecycle.json [hunk 6]
```diff
@@ -176,21 +161,24 @@
     {
       "@id": "forensic_action3",
       "@type": "ForensicAction",
-      "name": "imaged",
+      "name": "image",
```
Comment (casework) [inline examples/forensic_lifecycle.json]: Did we agree to change it to present tense?

File: examples/forensic_lifecycle.json [hunk 7]
```diff
@@ -252,32 +245,36 @@
...
+            "provenance_record12",
+            "forensic_action6"
+          ],
+          "location": "forensic_lab1",
+          "environment": "forensic_lab_computer1"
+        },
+        {
           "@type": "Process",
           "arguments": "log2timeline C:\\exams\\inbox\\case-123.img C:\\exams\\output\\case-123 --config C:\\plaso\\config.cfg --analysis chat_messages,windows_registries --output xlsx,pstorage --parsers sqlite/android_whatsapp,plist --log C:\\exams\\output\\case-123.log",
           "pid": 1234,
```
Comment (casework) [inline examples/forensic_lifecycle.json]: >     - I would suggest that the Process property bundle on forensic_action5 should actually go on the tool3 object as it is characterizing the tool used for the action.
        - Similarly, it seems like the PlasoArguments property bundle may be unnecessary. I may be misunderstanding it due to my lack of experience with Plaso but the configFile property seems like a reference to a config file used for that use of the tool (which should therefore be characterized on the tool object) and the analysisReport, storageFile and logFile properties seem to be referring to outputs of the action that should be captured and then referenced by the appropriate ProvenanceRecord.

The Process and PlasoArguments property bundles are characterizing an instance of using the tool. This information will change each time we use the tool and should therefore be on the action that is using the tool rather than the tool itself.

File: examples/forensic_lifecycle.json [hunk 8]
```diff
@@ -252,32 +245,36 @@
         "forensic_action5"
       ]
     },
+    // I think that forensic_action5 and forensic_action6 should be restructured
```
Comment (casework) [inline examples/forensic_lifecycle.json]: >    - I would strongly suggest that forensic_action5 and forensic_action6 should be restructured. Firstly, “executed” does not look like a useful forensic action to me. Just saying toolX was run does not really offer the information that should be provided. The forensic action should identify what sort of action the tool is performing (e.g., carve, query, extract, decrypt, parse, etc.) regardless of what specific tool was actually used. 

Agreed, we should change "executed" to something like "parsed" or "parse" (depending on the tense we agree on).

> Secondly, it does not make sense to me that forensic_action5 would take in both the android image and the sd card image. It seems there should be a separate forensic action for each of these.

Tools like plaso can take more than one input. A lot of parsers need to read files from both the phone and the associated sdcard in order to match attachments with messages.

Also, if we separate out the running of the parser from the framework, some parsers need to look at multiple files at once.

File: examples/location.json [hunk 9]
```diff
@@ -2,22 +2,21 @@
...
-        "schema:streetAddress": "20341 Whitworth Institute 405 N. Whitworth"
-      },
       "propertyBundle": [
         {
+          "@type": "SimpleAddress",
+          "locality": "Seattle",
+          "region": "WA",
+          "postalCode": "98052",
+          "street": "20341 Whitworth Institute 405 N. Whitworth"
+        },
```
Comment (casework) [inline examples/location.json]: Again, didn't realize we made our own property bundles. Does this mean we are no longer considering using an external ontology?

File: examples/message.json [hunk 10]
```diff
@@ -179,4 +179,4 @@
       ]
     },
   ]
-}
```
Comment (casework) [inline examples/message.json]: >         - We had never discussed needing “identifier”, “displayName” or “participants” properties for a thread. As such, they are not currently in the model. If a message thread is an actual data construct within messaging apps and they assign them properties like  “identifier”, “displayName” and “participants" then we should add them. If not, then I’m not sure why we would. “participants” seems particularly odd and like it would be especially simple to derive and unusual to capture explicitly on the thread.

They are their own constructs. Many apps keep conversation thread information separate from the messages themselves. They at least have an identifier and displayName, but can also have other stuff depending on the application. (eg. visibility, "secret/hidden", favorited/starred, topic, etc.)  The participants can sometimes be defined in this construct.

Comment (casework) [inline examples/message.json]: >         - Should thread have its own application ref property or is it adequate to link through the messages

Not sure. That's why it was better to have a relationship object pointing the application to all the elements that belonged to that application.  Not only do we need to associate messages and chat threads with applications but we also need to associate contacts, files, events, etc.

Comment (casework) [inline examples/message.json]: >     - We never discussed having the “message” property set of values be ordered. The model currently just has them as a set. It seems like the ordering should be easily derived from the timestamps on the messages referenced under the “message” property for the thread. Do messaging apps maintain actual data constructs for threads with explicit ordering? If so, then we should make the “message” property ordered.

Sometimes we can determine the order of messages without having timestamps. Its very rare, but it does happen.

Comment (casework) [inline examples/message.json]: >  On all Message traces, changed “sender” to “from” and “participant” to “to"
- On post1, removed account3 from “to” list as it seems odd to include the sender on the recipient list
- On message1, removed account1 from “to” list as it seems odd to include the sender on the recipient list

Note: This commit doesn't actually have any changes. I'm not sure where you made these changes.

The reason we have "sender" and "participant" instead of "to" and "from" is because sometimes we can't determine who sent the message, but we know the people who participated in the conversation. That's why the sender is also in the participant property. There are some cases where sender is null and we only have participants.

> - On message1, the “mimeType” and “size” properties are not valid on the Attachment property bundle. It looks like the concept/structure or attachments on messages needs to be further discussed and explored.

Sometimes, the message construct provides information about the attachment file such as "mimeType" and "size". These values are obviously not always the same as the actual file or may describe a file that has been deleted.

A really old idea that was previously disagreed on was to add the Attachment property bundle to the Trace describing the actual file. (If the file ended up not existing, we would have a Trace with just an "Attachment" property bundle.) And then create a relationship object pointing the message to the attachment object.

File: examples/raw_data.json [hunk 11]
```diff
@@ -8,38 +8,47 @@
...
-          "dataContainerType": "DataRange",
-          "content": "digital_photograph_thumbnail1_content"
+          "@type": "Hash",
+          "hashMethod": "MD5",
+          "hashValue": "3d137a188c1e82247b815209ce44af2c"
+        },
+        {
+          "@type": "File",
+          "magicNumber": "/9j/4AAQSkZ",
+          "mimeType": "image/jpg"
```
Comment (casework) [inline examples/raw_data.json]: This example has been already updated in the latest master. Please align these changes with that.

File: examples/sms_and_contacts.json [hunk 12]
```diff
@@ -7,47 +7,81 @@
     {
```
Comment (casework) [inline examples/sms_and_contacts.json]: All these changes have been mentioned in previous comments.

File: examples/file.json [hunk 13]
```diff
@@ -1,126 +1,209 @@
 {
```
Comment (casework) [inline examples/file.json]: A lot of these changes have already been applied in the latest master branch. Can you please align your changes with that?

File: examples/email.json [hunk 14]
```diff
@@ -7,36 +7,57 @@
...
-      "birthday": "1968-09-25T17:59:43.25Z",
-      "associatedAccount": [
-        "facebook_identity1",
-        "google_identity1",
-        "email_account1"
+      "propertyBundle": [
+        {
+          "@type": "SimpleName",
+          "givenName": "William",
+          "familyName": "Smith",
```
Comment (sbarnum) [inline examples/email.json]: I do not remember making a decision to use an external model.
I do not really have an objection to using schema.org for now but it is very limited in expressive coverage and UCO will need to work to flesh out more focused aspect property bundles.
For CASE, maybe we just use schema.org for now and extend/migrate to others once something better is concretely done. I will caveat that I am a bit worried about using schema.org in that their Person structure is not very clean. It munges a bunch of stuff into that structure including location info and a bunch of use-case-specific stuff. I think there is significant potential for that to cause implementer confusion.

File: examples/email.json [hunk 15]
```diff
@@ -73,7 +104,8 @@
               "willyROX@gmail.com"
           ],
```
Comment (sbarnum) [inline examples/email.json]: OK. I have modified the cardinality of accountLogin in the model to allow multiples.

File: examples/exif_data.json [hunk 16]
```diff
@@ -21,42 +21,23 @@
...
-        },
-        {
-          "@type": "RasterImage",
-          "format": "jpg",
-          "height": 12345,
-          "width": 12345,
+          "@type": "RasterPicture",
+          "pictureType": "jpg",
+          "pictureheight": 12345,
+          "picturewidth": 12345,
```
Comment (sbarnum) [inline examples/exif_data.json]: Yes, it is necessary. Due to explicit mapping to ontology (JSON-LD) each term needs to be unique. Cannot rely on implicit context scoping within a class.
Using something like "height" leads to significant semantic ambiguity. Is it describing the height of a picture in pixels, a person in meters, etc.?
Some properties are more generally and consistently useful across objects and can be named (and defined) more generally. Other properties differ in their definitions and application when applied on specific types of objects and they should be named accordingly to maintain integrity.

File: examples/exif_data.json [hunk 17]
```diff
@@ -103,11 +111,16 @@
...
-      "location": "location1"
+      "propertyBundle": [
+        {
+          "@type": "ActionRefs",
+          "instrument": "camera1",
+          "result": [
+            "digital_photograph1"
+          ],
+          "location": "location1"
+        }
```
Comment (sbarnum) [inline examples/exif_data.json]: We did discuss this extensively and in the end agreed to use ActionRefs. Eoghan made an explicit statement to this effect.
The argument for using ActionRefs is that there are clear use cases and users who will want/need to characterize interobject relationships for actions using external Relationships rather than embedded references. By utilizing a property bundle for these on Action we enable users who want to use refs to specify the ActionRefs property bundle as in-scope and the mechanism for doing this. At the same time we allow those who want to use Relationships to simply specify the ActionRefs property bundle as out of scope and specify Relationship as they mechanism for relating things.
If we moved the object relationships to be embedded properties directly on Action we make it impossible for the users/use-cases needing to use Relationship to do so without introducing two ways to do the same thing. Using a property bundle allows both variations in a non-ambiguous one way to do things manner.
The only real advantage to embedding the refs directly on Action is that it is slightly (VERY slightly) less verbose.
There should be no confusion or awkwardness to having some action properties in the bundle and some on the object. Many objects are done this way and there is a consistency to which properties are in the bundle and which are not. All of the properties in the bundle are relationship references to other objects. There are no such references left on Action.

File: examples/exif_data.json [hunk 18]
```diff
@@ -118,17 +131,21 @@
...
+        {
+          "performer": "examiner1",
+          "instrument": "annotator_tool1",
+          "result": [
+            "annotation1"
+          ],
+          "location": "forensic_lab1",
+          "environment": "forensic_lab_computer1"
+        }
+      ]
```
Comment (sbarnum) [inline examples/exif_data.json]: Good catch. It has been fixed

File: examples/message.json (hunk 1 above)
Comment (sbarnum) [inline examples/message.json]: The application seems an inherent and indelible property of the message in the way we are currently using message.
It seems that it would be awkward and unnecessarily verbose to make the application association of every message into separate Relationship objects.

File: examples/forensic_lifecycle.json (hunk 3 above)
Comment (sbarnum) [inline examples/forensic_lifecycle.json]: - The order should be derivable from the timestamps and the flow of Traces and ProvenanceRecords in and out of ForensicActions

- Yes, it could possible grow very large. The alternative to maintain clear and unambiguous relating of what content came from what investigation would be to use Relationships which would be WAY more content.

- Maybe we should make the object ref list into a property bundle so that adopters could choose which way to do it similar to ActionRefs.

- The Investigation object supports all cyber investigation use cases, not just DF. Use cases like incident response are unlikely to utilize ForensicAction and ProvenanceRecord for all of their information gathering and interpretation.

- We need a way to associate ANY relevant information with an investigation and not presume it will always be tied to a ProvenanceRecord.

- It is a perfectly valid decision (if consensus feels it makes sense) to specify that for CASE the “object” property of Investigation does not contain Traces or ProvenanceRecords and that they must be derived from the referenced ForensicActions. To me this seems like more of a processing burden than simply having the lengthy, complete explicit list. Either way, UCO will need to be able to reference any type of object in this field so any restriction would be solely for the CASE use of it.

File: examples/forensic_lifecycle.json (hunk 8 above)
Comment (sbarnum) [inline examples/forensic_lifecycle.json]: I think "parse" makes sense.

OK. I understand the multiple inputs for an action if it is being performed across that set of inputs.

File: examples/location.json (hunk 9 above)
Comment (sbarnum) [inline examples/location.json]: We did discuss having different property bundles for different sorts of location specifications (addresses, lat-long, GPS, etc.). I don't think we ever went into depth on whether there were available external options to leverage for this or not. I reviewed a few and didn't really see any clearly awesome options. In the absence of deep discussion and consensus on any great external options I created some basic property bundles in the model to use.

The big question is if we want to use an external one which external one to use
    - If use schema.org we would need to clearly limit it to PostalAddress.

The schema.org structure is inadequate for UCO that will require richer options but for CASE for now it could be limited to something like schema.org:PostalAddress

We just need to formally decide what to do.

File: examples/call_log.json [hunk 19]
```diff
@@ -43,11 +43,11 @@
           "callType": "mobile",
           "startTime": "2010-01-15T17:59:43.25Z",
           "endTime": "2010-01-15T18:30:41.25Z",
-          "sender": "phone_account1",
-          "participant": "phone_account2",
+          "from": "phone_account1",
+          "to": "phone_account2",
```
Comment (casework) [inline examples/call_log.json]: The reason we have "sender" and "participant" instead of "to" and "from" is because sometimes we can't determine who sent the message, but we know the people who participated in the conversation. That's why the sender is also in the participant property (the example should of had `phone_account1` as a participant as well). There are some cases where sender is null and we only have participants.

File: examples/email.json (hunk 14 above)
Comment (casework) [inline examples/email.json]: Sounds good. We should apply your change for now and worry about where we can apply external ontologies in the future.

File: examples/exif_data.json [hunk 20]
```diff
@@ -17,50 +17,27 @@
...
-          "data": "<base 64 encoded data of the file>",
-          "hash": [
-            {
-              "@type": "Hash",
-              "hashMethod": "SHA256",
-              "hashValue": "6b86b273ff34fce19d6b804eff5a3f5747ada4eaa22f1d49c01e52ddb7875b4b"
-            }
-          ]
+          "isDirectory": false,
+          "sizeInBytes": 35002
```
Comment (casework) [inline examples/exif_data.json]: We should keep the "FileSystem" property bundle on the Relationship object until we discuss otherwise and determine a new solution.

File: examples/exif_data.json (hunk 16 above)
Comment (casework) [inline examples/exif_data.json]: Agreed.

File: examples/exif_data.json (hunk 20 above)
Comment (sbarnum) [inline examples/exif_data.json]: hmm. I didn't take it off of the Relationship. 
I agree if there is a contained-within Relationship defined then the FileSystem bundle most appropriately goes on the Relationship.
Now I am worried that the git merge may have munged things I did not notice. :-(

File: examples/exif_data.json [hunk 21]
```diff
@@ -99,6 +76,33 @@
...
+      "@id": "relationship1",
+      "@type": "Relationship",
+      "source": "digital_photograph1",
+      "target": "device_partition3",
+      "kindOfRelationship": "contained-within",
+      "isDirectional": true,
+      "propertyBundle": [
+        {
+          "@type": "DiskPartition",
+          "partIndex": 3
```
Comment (casework) [inline examples/exif_data.json]: This relationship is telling me that the `digital_photograph1` is the 3rd partition in `device_paritition3`... 

The relationship that connects the photo to the partition should be the file system information (file path) that connects it.  *(The partition index  information connects the image to the partition.)*

File: examples/file.json [hunk 22]
```diff
@@ -13,12 +13,12 @@
...
           "sizeInBytes": 29,
-          "data": "Q0FTRSBpcyBhbiBhd2Vzb21lIG9udG9sb2d5IQ=",
+          "dataPayload": "Q0FTRSBpcyBhbiBhd2Vzb21lIG9udG9sb2d5IQ=",
           "hash": [
             {
               "@type": "Hash",
-              "hashMethod": "SHA256",
-              "hashValue": "6b51d431df5d7f141cbececcf79edf3dd861c3b4069f0b11661a3eefacbba918"
+              "hashMethod": "MD5",
+              "hashValue": "c99c56455e4362016016f251edab7c55"
```
Comment (casework) [inline examples/file.json]: This is changing the hashes back to md5. Please revert this change.

File: examples/file.json [hunk 23]
```diff
@@ -90,14 +90,24 @@
...
             "sizeInBytes": 23000,
-            "data": "<base 64 encoded data of the file>",
+            "dataPayload": "<base 64 encoded data of the file>",
             "hash": [
               {
                 "@type": "Hash",
-                "hashMethod": "SHA256",
-                "hashValue": "a665a45920422f9d417e4867efdc4fb8a04a1f3fff1fa07e998e86f7f7a27ae3"
+                "hashMethod": "MD5",
+                "hashValue": "56137a188c1e82247b815209ce33af2d"
```
Comment (casework) [inline examples/file.json]: Please keep it as SHA256.

File: examples/file.json [hunk 24]
```diff
@@ -124,14 +134,20 @@
...
           "sizeInBytes": 23000,
-          "data": "<base 64 encoded data of the file>",
+          "dataPayload": "<base 64 encoded data of the file>",
           "hash": [
             {
               "@type": "Hash",
-              "hashMethod": "SHA256",
-              "hashValue": "03ac674216f3e15c761ee1a5e255f067953623c8b388b4459e13f978d7c846f4"
+              "hashMethod": "MD5",
+              "hashValue": "45137a188c1e82247b815209ce33af2d"
```
Comment (casework) [inline examples/file.json]: Please keep it as SHA256.

File: examples/file.json [hunk 25]
```diff
@@ -90,14 +90,24 @@
...
+          },
+          {
+            "@type": "File",
+            "isDirectory": false,
+            "magicNumber": "/9j/ww==",
+            "mimeType": "image/jpg"
+          },
+          {
+            "@type": "Encoding",
+            "encodingMethod": "BASE64"
```
Comment (casework) [inline examples/file.json]: The encoding information should be placed in the relationship that connect the encoded file to the decoded file.

File: examples/exif_data.json (hunk 21 above)
Comment (sbarnum) [inline examples/exif_data.json]: ??
I don't think I changed this.
I thought you had created it this way. I interpreted it as digital_photograph1 is contained-within partition index 3 of device_partition3. I agree that last part always gave me pause.
It seemed like that DiskPartition bundle would more appropriately go on the relationship between device_partition3 and what is contained-within.
Is that what you are saying as well?
Given that such a relationship does not exist in the example it seems like that property bundle can just be tossed out.

File: examples/file.json (hunk 22 above)
Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.
I resolved all of the merge conflicts by choosing the sha256 values.
Not sure how some got missed.
Have I told you that git hates me? I think I have. ;-)

File: examples/file.json (hunk 23 above)
Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.

File: examples/file.json (hunk 24 above)
Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.

File: examples/file.json [hunk 26]
```diff
@@ -161,14 +177,21 @@
...
           "sizeInBytes": 54000,
-          "data": "<base 64 encoded data of the file>",
+          "dataPayload": "<base 64 encoded data of the file>",
           "hash": [
             {
               "@type": "Hash",
-              "hashMethod": "SHA256",
-              "hashValue": "5994471abb01112afcc18159f6cc74b4f511b99806da59b3caf5a9c173cacfc5"
+              "hashMethod": "MD5",
+              "hashValue": "bea5438a9521adc1b23a9117024bbb43"
```
Comment (casework) [inline examples/file.json]: Please keep it as SHA256.

File: examples/file.json [hunk 27]
```diff
@@ -161,14 +177,21 @@
...
             }
           ]
+        },
+        {
+          "@type": "Encryption",
+          "encryptionMethod": "AES",
+          "encryptionMode": "CBC",
+          "key": "whatsappKEY",
+          "iv": "whatsappIV"
```
Comment (casework) [inline examples/file.json]: Again, why are we adding the relationship information in the resulting file as well?
Adding it here, provides no extra information that cannot already be derived.

File: examples/file.json (hunk 26 above)
Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.

File: examples/file.json (hunk 25 above)
Comment (casework) [inline examples/file.json]: Didn't realize, the property bundle is also in the relationship object. Why is it both places?

File: examples/file.json (hunk 27 above)
Comment (sbarnum) [inline examples/file.json]: It does. It is not relationship information. It is information characterizing the trace itself.
See response to the comment above regarding the Encoding property bundle.

File: examples/file.json [hunk 28]
```diff
@@ -195,12 +218,12 @@
...
             "sizeInBytes": 546000,
-            "data": "<base 64 encoded data of the file>",
+            "dataPayload": "<base 64 encoded data of the file>",
             "hash": [
               {
                 "@type": "Hash",
-                "hashMethod": "SHA256",
-                "hashValue": "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92"
+                "hashMethod": "MD5",
+                "hashValue": "99bcf55577f19674034da613fe6664d9"
```
Comment (casework) [inline examples/file.json]: Please keep it as SHA256.

Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.

File: examples/file.json [hunk 29]
```diff
@@ -241,8 +265,8 @@
             "hash": [
               {
                 "@type": "Hash",
-                "hashMethod": "SHA256",
-                "hashValue": "8bb0cf6eb9b17d0f7d22b456f121257dc1254e1f01665370476383ea776df414"
+                "hashMethod": "MD5",
+                "hashValue": "5d41402abc4b2a76b9719d911017c592"
```
Comment (casework) [inline examples/file.json]: Please keep it as SHA256.

File: examples/file.json (hunk 25 above)
Comment (sbarnum) [inline examples/file.json]: That is explained in my comment directly above in they commit.
If it is still not clear I can explain verbally.

File: examples/file.json (hunk 29 above)
Comment (sbarnum) [inline examples/file.json]: I agree with keeping as sha256.

File: examples/forensic_lifecycle.json (hunk 3 above)
Comment (casework) [inline examples/forensic_lifecycle.json]: > Yes, it could possible grow very large. The alternative to maintain clear and unambiguous relating of what content came from what investigation would be to use Relationships which would be WAY more content.

I was thinking that the related object would be derived by tracing the forensic actions.. 

> Maybe we should make the object ref list into a property bundle so that adopters could choose which way to do it similar to ActionRefs.

Perhaps. If we put this information in a property bundle, we can have a handful of different property bundles to choose from based on your domain.

> The Investigation object supports all cyber investigation use cases, not just DF. Use cases like incident response are unlikely to utilize ForensicAction and ProvenanceRecord for all of their information gathering and interpretation.

True.. This would be a good argument for using property bundles to structure the investigation.

> We need a way to associate ANY relevant information with an investigation and not presume it will always be tied to a ProvenanceRecord.

I thought a long time ago we made it a rule that all Traces, Identities, Relationship objects, etc must be tied to a ProvenanceRecord.

File: examples/exif_data.json (hunk 17 above)
Comment (casework) [inline examples/exif_data.json]: Agreed. But perhaps we should change the name to something more meaningful. Maybe "ActionParameters"?

Comment (sbarnum) [inline examples/exif_data.json]: If we can come up with a better name I am fine with changing.
I do not think ActionParameters is appropriate as the meaning of action parameters has a different and very specific meaning in use (the parameters utilized in carrying out an action.
ActionRefs are broader than that specific meaning.

File: examples/forensic_lifecycle.json (hunk 7 above)
Comment (casework) [inline examples/forensic_lifecycle.json]: > They are out of place on the Action as they are characterizing details of the tools referenced in the “instrument” property. Having them be property bundles on the Action is ambiguous as they are not characterizing the action itself but rather another property of the action (what instrument was used to carry out the action).

True, but perhaps the solution is not to take the process property bundle out of the action, but to remove the "instrument" property out of ActionRef and have some type of "ToolParameters" property bundle... I'm just brainstorming.. lets talk about this more later.

> Each instantial use of a tool should include all the details for that use and then any Actions that used that instance can reference it. This is the way Tool is intended to be utilized. Each instantial use could also potentially define an explicit Relationship to a more general entry for the tool or simply rely on the implicit nature of its properties.

So, the name "Tool" is obviously misleading then...

File: examples/location.json (hunk 9 above)
Comment (casework) [inline examples/location.json]: Lets just use our own ontology for now.

File: examples/message.json (hunk 1 above)
Comment (casework) [inline examples/message.json]: True.

File: examples/call_log.json (hunk 19 above)
Comment (casework) [inline examples/call_log.json]: Agreed to have 3 properties: "to", "from", and "participant"

File: examples/exif_data.json (hunk 20 above)
Comment (casework) [inline examples/exif_data.json]: Caused by git merge issue.

File: examples/exif_data.json (hunk 21 above)
Comment (casework) [inline examples/exif_data.json]: Agreed to remove the relationship of the android_image pointing to disk_partition from the example.

File: examples/forensic_lifecycle.json (hunk 6 above)
Comment (casework) [inline examples/forensic_lifecycle.json]: Agreed to keep present tense for now. 

TODO: Bring this up with the group and have a final vote.
